backend/routes/userNews.js

```diff
@@ -161,12 +173,104 @@ router.put("/:newsId/bookmarked", async (req, res) => {
 
       personalNews.push(newArticle[0]);
     }
+
     res.json(personalNews);
   } catch (error) {
     res.status(500).json({ error: "Error updating metadata" });
   }
 });
 
+// get a personalized feed
+router.post("/personalized", async (req, res) => {
+    const newNews = await prisma.userNewsCache.findMany({
+      where: { userId: req.session.userId },
+      orderBy: {
+        id: "desc",
+      },
+    });
+
+    const personalNews = [];
+    // there are entries found
+    for (const art of newNews) {
+      const newArticle = await prisma.news.findMany({
+        where: { id: art.newsId },
+      });
+
+      newArticle[0].score = art.score;
+      newArticle[0].bookmarked = art.bookmarked;
+
+      personalNews.push(newArticle[0]);
+    }
+
+    try {
+         // sort news based on engagement scores
+    personalNews.sort((a, b) =>  (b['score']) - (a['score']))
+    const top10 = personalNews.slice(0,10); // top 10 articles with highest engagement scores
```

Comment:
are we limiting userNewsCache to a max number of articles? like 30? If so then this part is fine